compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1770,82 +1773,97 @@ private Result seekInAccessExpression(ParserRuleContext currentCtx, int lookahea
+                        ParserRuleContext.CLOSE_BRACKET };
+                break;
             case LET_EXPR_LET_VAR_DECL:
-                next = new ParserRuleContext[] { ParserRuleContext.COMMA, ParserRuleContext.BINARY_OPERATOR,
+                alternatives = new ParserRuleContext[] { ParserRuleContext.COMMA, ParserRuleContext.BINARY_OPERATOR,
                         ParserRuleContext.DOT, ParserRuleContext.ANNOT_CHAINING_TOKEN,
                         ParserRuleContext.OPTIONAL_CHAINING_TOKEN, ParserRuleContext.CONDITIONAL_EXPRESSION,
                         ParserRuleContext.XML_NAVIGATE_EXPR, ParserRuleContext.MEMBER_ACCESS_KEY_EXPR,
-                        ParserRuleContext.IN_KEYWORD, ParserRuleContext.ARG_LIST_START };
-                return seekInAlternativesPaths(lookahead, currentDepth, currentMatches, next, isEntryPoint);
+                        ParserRuleContext.IN_KEYWORD };
+                break;
             case LET_CLAUSE_LET_VAR_DECL:
-                next = new ParserRuleContext[] { ParserRuleContext.COMMA, ParserRuleContext.BINARY_OPERATOR,
+                alternatives = new ParserRuleContext[] { ParserRuleContext.COMMA, ParserRuleContext.BINARY_OPERATOR,
                         ParserRuleContext.DOT, ParserRuleContext.ANNOT_CHAINING_TOKEN,
                         ParserRuleContext.OPTIONAL_CHAINING_TOKEN, ParserRuleContext.CONDITIONAL_EXPRESSION,
                         ParserRuleContext.XML_NAVIGATE_EXPR, ParserRuleContext.MEMBER_ACCESS_KEY_EXPR,
-                        ParserRuleContext.LET_CLAUSE_END, ParserRuleContext.ARG_LIST_START };
-                return seekInAlternativesPaths(lookahead, currentDepth, currentMatches, next, isEntryPoint);
+                        ParserRuleContext.LET_CLAUSE_END };
+                break;
             case QUERY_EXPRESSION:
-                next = new ParserRuleContext[] { ParserRuleContext.BINARY_OPERATOR, ParserRuleContext.DOT,
+                alternatives = new ParserRuleContext[] { ParserRuleContext.BINARY_OPERATOR, ParserRuleContext.DOT,
                         ParserRuleContext.ANNOT_CHAINING_TOKEN, ParserRuleContext.OPTIONAL_CHAINING_TOKEN,
                         ParserRuleContext.CONDITIONAL_EXPRESSION, ParserRuleContext.XML_NAVIGATE_EXPR,
-                        ParserRuleContext.MEMBER_ACCESS_KEY_EXPR, ParserRuleContext.QUERY_PIPELINE_RHS,
-                        ParserRuleContext.ARG_LIST_START };
-                return seekInAlternativesPaths(lookahead, currentDepth, currentMatches, next, isEntryPoint);
+                        ParserRuleContext.MEMBER_ACCESS_KEY_EXPR, ParserRuleContext.QUERY_PIPELINE_RHS };
+                break;
             default:
                 if (isParameter(parentCtx)) {
-                    next = new ParserRuleContext[] { ParserRuleContext.CLOSE_PARENTHESIS,
+                    alternatives = new ParserRuleContext[] { ParserRuleContext.CLOSE_PARENTHESIS,
```

Comment:
Will the above logic be simplified, if we take the common things out, similar to the second switch-case?

Reply:
Common alternatives are located in the middle of array. No of start and end alternatives are different for each ctx. So are we going to add the common ones at the end?

Reply:
btw `MEMBER_ACCESS_KEY_EXPR` is missing as an alternative under `COMPUTED_FIELD_NAME` ctx. I think it's a mistake